gizmo.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+GIZMO Challenge
+
+@author: jenni.saaristo@helsinki.fi
+@version: 20210319
+@notes: Funfunfun!
+"""
+
+import numpy as np
+import pandas as pd
+
+
+# -------------------- Exercise 1 --------------------
+
+def hello(name, country='Finland'):
+    print('Hello ' + name + ', how are things in ' + country + '?')
+
+# -------------------- Exercise 2 --------------------
+
+def spell(name):
+    out = ''
+    for i in range(len(name)):
+        out = out + name[i] + '.'
+    print(out[:-1])
+
+# -------------------- Exercise 3 --------------------
+
+def relative_path(ids):
+    base = ['./subjects/mock_recording_','.rec']
+    names = []
+    
+    if type(ids) == str:
+        ids = [ids]
+    for s in ids:
+        names.append(base[0] + s + base[1])
+    
+    return names
+
+# ------------------- Exercise 4–7 --------------------
+
+class Gizmo:
+    
+    def __init__(self, name):
+        self.name = name
+    
+    def speak(self):
+        print(self.name)
+
+# ------------------- Exercise 8-9 --------------------
+
+def multiplication_table(zero_out_multiples=None):
+    """Times table with optional dropping of multiples
+    
+    Returns the times table for integers up to 12, and optionally
+    zeroes the multiples of the given int argument.
+    
+    Parameters
+    ----------
+    zom : int, optional
+        The integer to drop the multiples of (zero_out_multiples)
+    
+    Returns
+    -------
+    tab : numpy array
+        The times table as an array
+        
+    """
+
+    v = np.arange(1,13)
+    tab = np.outer(v,v)
+    if type(zero_out_multiples) == int:
```

Comment:
Interesting way to check that `zero_out_multiples` is not `None`. Personally I recommend:

```python
if zero_out_multiples is not None:
```

Reply:
Ahh, true! But I kinda also wanted to catch floats etc.

Reply:
But a float should probably raise an error, not be silently ignored